os/net/socket/net_vfcntl.c

```diff
@@ -99,7 +99,7 @@
 int net_vfcntl(int sockfd, int cmd, va_list ap)
 {
 
-	FAR struct socket *sock = (struct socket *)get_socket(sockfd);
+	FAR struct socket *sock = (struct socket *)get_socket(sockfd, getpid());
```

Comment:
I think you can get pid inside get_socket()

Reply:
yeah we can get pid inside the get_socket(), but the thing is many time process gets changed. like in event_callback(), get_socket() is called and at that time context switch was happening so could not fetch the correct socketlist this is why i am passing the pid from the function itself. for event_callback() i am passing the pid after taking it from netconn.

Reply:
I can't see the case you told "and at that time context switch was happening so could not fetch the correct socketlist this is why i am passing the pid from the function itself. for event_callback() i am passing the pid after taking it from netconn".
Did you upload it in different PR?

Reply:
pid was getting changed either in event_callback() or lwip_setsockopt_impl() or in both. don't remember exactly. but i am sure in at least one of the mentioned function, pid was getting changed. this is why i had to pass pid as argument to fetch correct socketlist.

no, i didn't upload it in the PR of my forked repo that's why i was getting error.